core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -191,9 +191,20 @@ private[spark] class Executor(
    */
   private val HEARTBEAT_INTERVAL_MS = conf.get(EXECUTOR_HEARTBEAT_INTERVAL)
 
+  /**
+   * Interval to poll for executor metrics, in milliseconds
+   */
+  private val METRICS_POLLING_INTERVAL_MS = conf.get(EXECUTOR_METRICS_POLLING_INTERVAL)
```

Comment:
seems that we have a config for logging executor metrics that before was basedon heartbeat, now your change is adding another config to have the metrics based on a new interval. That's sound like a lot of redundant metrics. Maybe we want to simplify configs and remove the previous one and then default this one to heartbeat for simplification.

Reply:
When executor metrics were introduced, there was a config introduced, spark.eventLog.logStageExecutorMetrics.enabled. However, that only controlled whether aggregated metrics (peaks per stage per executor) were written to event logs; it did not control polling. Regardless of the value of spark.eventLog.logStageExecutorMetrics.enabled, executor metrics were polled and sent in the executor heartbeat. The polling was done in the reportHeartBeat function, so was done at the frequency of the heartbeat.
In order to have faster polling, which is the whole point of this change, it is necessary to have a config specifying how frequently to poll, so there is no redundant config, if that is what you mean. It should not be tied to the executor heartbeat frequency, since the use case is we don't want faster heartbeats, only faster polling.
And there is no redundant polling for metrics, if that is what you mean. If spark.executor.metrics.pollingInterval is not set, then we continue to poll in reportHeartBeat, as before. There are not two separate polling threads; there is only ever one thread per executor that polls executor metrics.

Reply:
After your change being merged for someone to get executor metrics they need to consider 4 configs. 1) spark.eventLog.logStageExecutorMetrics.enabled 2)"spark.eventLog.logStageExecutorProcessTreeMetrics.enabled" 3)"spark.executor.heartbeatInterval" 4)"spark.executor.metrics.pollingInterval"
Number 2 is totally different and we can't consolidate. However I feel like we can say that whenever number 4 is set then user definitely want number one too. So They don't need to set both.

Reply:
3 is an independent thing. The problem before this change is that the only way to get faster polling is to increase 3, which is abusing it. Now, the user doesn't need to do anything to 3 at all (unless for some independent reason they want to change the heartbeat interval). They tune 4 instead. So I have not actually increased the number of configs the user needs to touch.
I do agree that the only cases I can think of where a user would want to set 4 are when they want to set 1 to true as well, but then the same can be said of 2. When a user sets 2 to true, they would want to set 1 to true as well. Nevertheless, I don't think we want to couple the configs; I think it is still necessary to have separate configs and let the user set them.

Reply:
3 is still in use of 1 and 2 and if somebody don't want to use 4 they will need to tune 3. 
And it was bad that we didn't document the relation between (2) and (1). 
If you like you can comment it some where at least that for this change the user need to set both (1) and (4). This is to avoid users file issues.

Reply:
I'll make a follow-up docs PR to add 1, 2 and 4 to the Configuration page and explain their use.
If there are no code changes needed, I don't want to upload another commit and triggering another lengthy test run.

Reply:
Up to to you. There can be any more changes here, since there could be other people that want to comment here.